Release note for 1.8.0 (#3567)
line/armeria: site/src/pages/release-notes/1.8.0.mdx

---
date: 2021-05-14
---

## 🌟 New features

> - You can now create a <type://HealthChecker> that invokes a health checker repetitively. #3392 #3406

Reviewer: How about `... that updates its healthiness periodically.`?

  ```java
  HealthChecker.of(() -> {
      HealthCheckStatus status = ...;
      CompletableFuture.completedFuture(status);
  }, Duration.ofSeconds(5));
  ```

> - You can now create a propagating <type://ContextAwareExecutorService> and

Reviewer: ```suggestion
- You can now create the <type://RequestContext> propagating <type://ContextAwareExecutorService> and
```

Author: `RequestContext.makeContextPropagating()` actually returns a `ExecutorService` or `ScheduledExecutorService`

  <type://ContextAwareScheduledExecutorService> using <type://RequestContext#makeContextPropagating(ExecutorService)?full>
  and <type://RequestContext#makeContextPropagating(ScheduledExecutorService)?full>.
  The executors propagate a <type://RequestContext> to submitted tasks when a submission thread is 
  a request-context-aware thread. #3483
  ```java
  ExecutorService myExecutor = Executors.newSingleThreadExecutor();
  ExecutorService contextAwareExecutor = RequestContext.makeContextPropagating(myExecutor);
  // Make sure the current thread is a request-context-aware thread
  RequestContext ctx = RequestContext.current();
  contextAwareExecutor.execute(() -> {
      assert RequestContext.current() == ctx;
  });
  ```
- You can now set a maximum allowed length of a server response in the request level
  using <type://WebClientRequestPreparation#maxResponseLength(long)>. #3565
  ```java
  WebClient client = ...;
  client.prepare()
        .maxResponseLength(maxResponseLength)
        ...
        .execute();
  ```
- You can now add or set HTTP headers and query params with `java.util.Map`. #3499 #3502
  ```java
   HttpHeaders
      .builder()

>       .add(Map.of("foo", "1"));

Reviewer: ```suggestion
      .add(Map.of("foo", "1"))
```

      .set(Map.of("bar", "2"));
  
  QueryParams.builder()

>              .add(Map.of("page", "1"));

Reviewer: ```suggestion
             .add(Map.of("page", "1"))
```

             .set(Map.of("size", "2"));
  ```
- You can now disable DNS query metrics using <type://DnsResolverGroupBuilder#disableDnsQueryMetrics()>.
  #3368 #3552
  ```java

>   ClientFactory.builder()
>              .domainNameResolverCustomizer(customizer -> {

Reviewer: ```suggestion
  ClientFactory.builder()
               .domainNameResolverCustomizer(customizer -> {
```
